### PR TITLE
documentation: Flask-Wiki integration

### PR DESCRIPTION
**Pull request by jma**
This new module will be used to create the public documentation
dedicated to the users and the librarians.

* Adds Flask-Wiki dependency.
* Adds Flask-Wiki default configuration.
* Creates permissions to limit the wiki edition to a dedicated user.
* Fixes bootstrap toast js and css to avoid external module collusion.
* Moves head.html template code to the main page to be able to overwide
the css block.
* Includes Flask-Wiki translations.
* Removes useless code in the Dockerfile.
* Adds specific css for markdown figures.
* Creates fixtures for the dev instance and for the tests.

Co-Authored-by: Johnny Mariéthoz <Johnny.Mariethoz@rero.ch>
Co-Authored-by: Igor Milhit <igor.milhit@rero.ch>

## Why are you opening this PR?

- To implement a simple solution to create the public pages documentation.

## How to test?

- pipenv run bootstrap; pipenv run setup
- go to the help page
- login with the editor account account
- create you first documentation page
- save
- try to upload a new image file
- integrate this new image link in a help page

## Code review check list

- [x] Commit message template compliance.
- [x] Commit message without typos.
- [x] File names.
- [x] Functions names.
- [x] Functions docstrings.
- [x] Unnecessary commited files?

### CODE REVIEW

File: Pipfile [hunk 1]
```diff
@@ -77,4 +78,3 @@ python_version = "3.6"
 test = "python setup.py test"
 build_sphinx = "python setup.py build_sphinx"
 dev = "pytest --no-cov -vs tests"
-setup = "./scripts/setup"
```
Comment (blankoworld) [inline Pipfile]: If you delete `setup` you should change `./scripts/setup` when it asks user to use `pipenv run setup` and suggest `pipenv run ./scripts/setup` instead.
Same thing in `INSTALL.rst` file.

File: Dockerfile [hunk 2]
```diff
@@ -36,4 +36,4 @@ ARG UI_TGZ=""
 
 ENV INVENIO_COLLECT_STORAGE='flask_collect.storage.file'
 
-RUN ./scripts/bootstrap --deploy ${UI_TGZ} && rm -fr ui/node_modules
+RUN pipenv run bootstrap --deploy ${UI_TGZ} && rm -fr ui/node_modules
```
Comment (blankoworld) [inline Dockerfile]: I think we can delete `rm -fr ui/node_modules` as we don't have `ui` anymore. Do you confirm that?

File: scripts/bootstrap [hunk 3]
```diff
@@ -22,7 +22,7 @@ SUCCESS_COLOR='\033[1;97;42m'   # Bold + white + green background
 ERROR_COLOR='\033[1;97;41m'     # Bold + white + red background
 
 PROGRAM=`basename $0`
-
+PREFIX=""
```
Comment (blankoworld) [inline scripts/bootstrap]: Don't forget to delete this line :wink:

File: scripts/console [hunk 4]
```diff
@@ -16,6 +16,52 @@
 # You should have received a copy of the GNU Affero General Public License
 # along with this program. If not, see <http://www.gnu.org/licenses/>.
 
+NC='\033[0m'                    # Default color
+INFO_COLOR='\033[1;97;44m'      # Bold + white + blue background
+SUCCESS_COLOR='\033[1;97;42m'   # Bold + white + green background
+ERROR_COLOR='\033[1;97;41m'     # Bold + white + red background
+
+PROGRAM=`basename $0`
+PREFIX=""
```
Comment (blankoworld) [inline scripts/console]: Don't forget to delete this line :wink:

File: scripts/server [hunk 5]
```diff
@@ -16,18 +16,63 @@
 # You should have received a copy of the GNU Affero General Public License
 # along with this program. If not, see <http://www.gnu.org/licenses/>.
 
+NC='\033[0m'                    # Default color
+INFO_COLOR='\033[1;97;44m'      # Bold + white + blue background
+SUCCESS_COLOR='\033[1;97;42m'   # Bold + white + green background
+ERROR_COLOR='\033[1;97;41m'     # Bold + white + red background
+
+PROGRAM=`basename $0`
+PREFIX=""
```
Comment (blankoworld) [inline scripts/server]: Don't forget to delete this line :wink:

File: scripts/update [hunk 6]
```diff
@@ -16,10 +16,54 @@
 # You should have received a copy of the GNU Affero General Public License
 # along with this program. If not, see <http://www.gnu.org/licenses/>.
 
+NC='\033[0m'                    # Default color
+INFO_COLOR='\033[1;97;44m'      # Bold + white + blue background
+SUCCESS_COLOR='\033[1;97;42m'   # Bold + white + green background
+ERROR_COLOR='\033[1;97;41m'     # Bold + white + red background
+
+PROGRAM=`basename $0`
+PREFIX=""
```
Comment (blankoworld) [inline scripts/update]: Don't forget to delete this line :wink:

File: scripts/update [hunk 7]
```diff
@@ -16,10 +16,54 @@
...
+  colored_msg "${SUCCESS_COLOR}" "SUCCESS" "${1}"
+}
+
+# Displays program name
+msg "PROGRAM: ${PROGRAM}"
+
 set -e
 
-script_path=$(dirname "$0")
-./"$script_path"/bootstrap
```
Comment (blankoworld) [inline scripts/update]: In my opinion you should prefer these lines.

**Comment by jma** [comment]
> In your commit message, at the point "How to test ?" , could you please add a word about the "setup" needed after the bootstrap ?
> Because without a setup , the new stuff (editor role,...) are not created in the local instance. :-\
> Thank you !

Oups, sorry. It is done.

**Comment by jma** [comment]
> After a test, the language switch (french <-> english) doesn't work.
> It stays on the french version.
> Any idea ?

Yes I works at it supposed to be. If you are in a page without suffixes (_en, _fr, etc.), it will display the page in the language interface if it is exists else the default. You can try it for the documentation root. Only english, french and default exists. The default is the fallback. But in the documentation pages the links are done explicitly in the page language i.e. in the french home page the links contains suffixes thus once you are in the sub pages you cannot switch the language unless the links are changed to default. It is up to the editor.

**Pull request merged**